io/io/src/TBufferFile.cxx

```diff
@@ -2518,7 +2518,7 @@ void TBufferFile::WriteObjectClass(const void *actualObjectStart, const TClass *
          // A warning to let the user know it will need to change the class code
          // to  be able to read this back.
          if (!actualClass->HasDefaultConstructor(kTRUE)) {
-            Warning("WriteObjectClass", "since %s has no public constructor\n"
+            Warning("WriteObjectAny", "since %s has no public constructor\n"
```

Comment:
Indeed, this is the intent.  WriteObjectClass is a really internal function and it made more sense to use the outer interface name.